### PR TITLE
Fix client-side validation of integer literals for number fields

### PR DESCRIPTION
**Pull request by liggitt**
Resolves an error trying to validate data containing an integer literal, unmarshaled using the integer-preserving https://github.com/kubernetes/apimachinery/blob/v0.18.0/pkg/util/json/json.go#L41-L43 method, against a schema field defined to accept numbers (floats or ints)

See https://swagger.io/docs/specification/data-models/data-types/#numbers:

> OpenAPI has two numeric types, number and integer, where number includes both integer and floating-point numbers.

### CODE REVIEW

File: pkg/util/proto/testdata/swagger.json [hunk 1]
```diff
@@ -6,6 +6,621 @@
   },
   "paths": {},
   "definitions": {
+    "io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceColumnDefinition": {
```
Comment (apelisse) [inline pkg/util/proto/testdata/swagger.json:9]: Was this copied from somewhere or generated from something in that repo?

Comment (apelisse) [reply pkg/util/proto/testdata/swagger.json:9]: Or did you type all these characters with your keyboard?

Comment (liggitt) [reply pkg/util/proto/testdata/swagger.json:9]: copied from k/k master swagger.json

**Pull request merged**